core/output/slack.js

```diff
@@ -206,6 +205,68 @@ init = function(token, callback) {
 	});
 },
 
+startPingPongTimer = function(teamId) {
+	if (!Date.now) {
+    	Date.now = function() {
+			return new Date().getTime();
+		}
+	}
+
+	var team = exports.config.slack_teams[teamId],
+		messageId = ++team.ping_id;
+
+	if (!messageId) {
+		messageId = 0;
+		team.ping_id = messageId;
+	}
+
+	team.timeSincelastMessageRecieved = Date.now();
+	setTimeout(() => {
+		if (Date.now() - team.timeSincelastMessageRecieved >= 60000 && messageId === team.ping_id) {
+			console.debug('sending ping');
+			sendPing(teamId);
+		}
+	}, 60000);
+},
+
+sendPing = function(teamId) {
+	var socket = sockets[teamId],
+		slackTeams = exports.config.slack_teams,
+		team = slackTeams[teamId],
+		pongId = ++team.pong_id;
+
+	if (!pongId) {
+		pongId = 0;
+		team.pong_id = pongId;
+	}
+
+	if (socket != null) {
+		var body = {
+		'id': team.pong_id,
+		'type':'ping',
+		'timestamp': Date.now()
+		};
+		body = JSON.stringify(body);
+		socket.send(body);
+
+		setTimeout(() => {
+			if (team.pong_id === pongId) {
+				console.debug("terminating connection as socket failed to respond to ping request.")
+				sockets[teamId].terminate();
+			}
+		}, 30000);
```

Comment:
30 seconds was just what seemed like a reasonable timeout time I will document at top of module